File: sdks/java/src/support/tests/UFG.java
package support.tests;

import com.applitools.eyes.RectangleSize;
import com.applitools.eyes.Region;
import com.applitools.eyes.selenium.StitchMode;
import com.applitools.eyes.selenium.fluent.Target;
import org.openqa.selenium.By;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;
import support.BaseSetup;

public class UFG extends BaseSetup {

    @BeforeClass
    public void classSetup() {
        UFG = true;
    }

    @Test
    public void window(){
        driver.get("https://applitools.github.io/demo/TestPages/FramesTestPage/");
        eyes.open(driver, "Eyes Selenium SDK - Classic API", "CheckWindowDefaultFully_VG", new RectangleSize(700, 460));
        eyes.check(Target.window());
        eyes.close(true);
    }

    @Test
    public void region() {
        driver.get("https://applitools.github.io/demo/TestPages/FramesTestPage/");
        eyes.open(driver, "Eyes Selenium SDK - Fluent API", "TestCheckRegionByCoordinates_Fluent_VG", new RectangleSize(700, 460));
        eyes.check(Target.region(new Region(50, 70, 90, 110)));
        eyes.close(true);
    }

    @Test
    public void frame() {
        eyes.setBranchName("universal-sdk");
        driver.get("https://applitools.github.io/demo/TestPages/FramesTestPage/");
        eyes.open(driver, "Eyes Selenium SDK - Classic API", "TestCheckFrame_VG", new RectangleSize(700, 460));
        eyes.check(Target.frame(By.cssSelector("[name=\"frame1\"]")));
        eyes.close(true);
    }
}

File: sdks/python/tests/support/UFG.py
import pytest
from applitools.selenium import (Region, Target, VisualGridRunner)


@pytest.fixture(scope="function")
def eyes_runner_class():
    return VisualGridRunner(10)


def window(driver, eyes):
    driver.get("https://applitools.github.io/demo/TestPages/FramesTestPage/")
    conf = eyes.get_configuration()
    conf.test_name = "CheckWindowDefaultFully_VG"
    conf.app_name = "Eyes Selenium SDK - Classic API"
    conf.viewport_size = {"width": 700, "height": 460}
    eyes.set_configuration(conf)
    eyes.open(driver)
    eyes.check(Target.window())
    eyes.close(raise_ex=True)


def region(driver, eyes):
    driver.get("https://applitools.github.io/demo/TestPages/FramesTestPage/")
    conf = eyes.get_configuration()
    conf.test_name = "TestCheckRegionByCoordinates_Fluent_VG"
    conf.app_name = "Eyes Selenium SDK - Fluent API"
    conf.viewport_size = {"width": 700, "height": 460}
    eyes.set_configuration(conf)
    eyes.open(driver)
    eyes.check(Target.window().region(Region(50, 70, 90, 110)))
    eyes.close(raise_ex=True)


def frame(driver, eyes):
    driver.get("https://applitools.github.io/demo/TestPages/FramesTestPage/")
    conf = eyes.get_configuration()
    conf.branch_name = "universal-sdk"
    conf.test_name = "TestCheckFrame_VG"
    conf.app_name = "Eyes Selenium SDK - Classic API"
    conf.viewport_size = {"width": 700, "height": 460}
    eyes.set_configuration(conf)
    eyes.open(driver)
    eyes.check(Target.window().frame("frame1"))
    eyes.close(raise_ex=True)
